test/metabase/api/dataset_test.clj

```diff
@@ -227,6 +230,75 @@
             (data-perms/set-database-permission! (perms-group/all-users) (mt/id) :perms/create-queries :no)
             (do-test)))))))
 
+(deftest native-query-with-truncated-column-alias
+  (testing "nested native query with long truncated column alias (metabase#47584)"
+    (mt/with-temp-copy-of-db
```

Comment:
Why is this copy needed?

Reply:
To be honest I cargo-culted this from another similar test in this file, but I assumed it was needed to prevent polluting the db with changes made in the `with-temp` form, below? It's confusing that the `with-temp-copy-of-db` and `with-temp` forms are separated here, but I needed to move this form to the top so the query-building forms in the `let` get the new temp db id when they call `(mt/id)`, etc.

Reply:
Removed in latest in commit. Good catch!